packages/dev-server/src/MetroDevServer.ts

```diff
@@ -138,7 +141,29 @@ export async function bundleAsync(
   };
 
   try {
-    return await Promise.all(bundles.map((bundle: BundleOptions) => buildAsync(bundle)));
+    return await Promise.all(
+      bundles.map(async (bundle: BundleOptions) => {
+        const bundleOutput = await buildAsync(bundle);
+        const shouldBuildHermesBundle = await shouldBuildHermesBundleAsync(
+          projectRoot,
+          bundle.platform
+        );
+
+        if (shouldBuildHermesBundle) {
+          options.logger.info({ tag: 'expo' }, `💿 Building hermes bundle for ${bundle.platform}`);
```

Comment:
I wonder if we should use a bit more general terms here like "bytecode bundle" or "hermes bytecode bundle", calling it a "hermes bundle" reminds me of when RN called the Metro bundler the "React Native packager". If in the future we want to support other engines, keeping it agnostic from the start could be a good thing. CC @brentvatne @ide

Reply:
prefer "hermes bytecode bundle" if the name is not too long even in other places.

btw, v8 use snapshot or code caching (w/ slightly different concept)

Reply:
Perhaps "Hermes bytecode for the bundle" is the most accurate description.

However, it is important that the Expo runtime does not have the concept of the bundle. Expo is like the web, not React Native. The web does not have a concept of a bundle. You can create bundles with tools like Webpack and the browser just treats it like a JS file.

Similarly, Expo apps don't have bundles. They have a collection of assets and one "launch asset", which is the first asset we run to launch an update. The launch asset currently needs to be JS but it could also be HBC or another format in the future. We just need to support running those new formats. Additionally, the launch asset does not need to be a bundle. It could be an unbundled JS file. Expo does not care because Expo is like the web.

Reply:
"Hermes bytecode for the bundle"  sounds clear in this context.
Let me update the code to something like "Building Hermes bytecode for the bundle - platform[android]".
If there are any comments, could update further.